app/assets/javascripts/components/DisciplineScatterPlot.js

```diff
@@ -110,6 +114,7 @@ DisciplineScatterPlot.propTypes = {
   seriesData: PropTypes.array.isRequired, // array of JSON event objects.
   titleText: PropTypes.string, //discipline dashboard makes its own title
   measureText: PropTypes.string.isRequired,
+  studentChart: PropTypes.bool.isRequired, //flag to indicate the chart is displaying individual student data
```

Comment:
Can you accomplish this with the current API, by passing different `tooltip`s?

This change would modify the generic scatterplot component to now have code that's specific to the profile page and dashboard page.  It looks like you could have them pass in the `tooltip` they need, and then this component stays generic.

Reply:
@kevinrobinson you're fast on the draw! I was actually looking at this and I agree that it's fine to pass a function here instead of a flag. At the time I thought there might be other reasons to distinguish behavior based on whether it was for a student or school.

Reply:
👍 also the test failure https://travis-ci.org/studentinsights/studentinsights/builds/480544138?utm_source=github_status&utm_medium=notification seems wacky, you can try just restarting it and if it still fails ping me and I can help debug:

![image](https://user-images.githubusercontent.com/1056957/51275362-c0674a80-199f-11e9-9d96-10183fb7daef.png)

Reply:
@kevinrobinson the old turn-it-off-turn-it-on-again trick does it again

Reply:
cool, yeah there's got to be a race condition in that spec related to testing scrubbing data sent to Rollbar, but not sure where.  So if it's happening .05% of the time then not so bad but yeah flaky tests are no bueno...